lib/csdl2openapi.js

```diff
@@ -1487,7 +1487,10 @@ module.exports.csdl2openapi = function (
         const updName = allName + '-update';
         const isKey = keyMap(type);
         const required = Object.keys(isKey);
-        const allProperties = {};
+        const metadata = csdl.$Version > '4.0' ? 
+            { '@id':{ type: "string", readOnly: true }, '@etag': { type: 'string', readOnly: true, example: "W/\"08D7D40891852C76\"" }, '@type':{ type: 'string', readOnly: true } } : 
+            { '@odata.id':{ type: "string", readOnly: true }, '@odata.etag': { type: 'string', readOnly: true, example: "W/\"08D7D40891852C76\"" }, '@odata.type':{ type: 'string', readOnly: true } };
+        const allProperties = { ...metadata };

```

Comment:
Please remove `readOnly:true`, the "all" structure is exclusively used for read operations.

`@id` is only necessary if not all key properties are sent, so typically it is not there - please remove.

`@type` only needs to be sent if the structured type has a `$BaseType`, please make conditional.

`@type` is necessary in the `creProperties` if the structured type has a `$BaseType`, please conditionally add it there, too.